activerecord/lib/rails/generators/active_record/model/model_generator.rb

```diff
@@ -43,9 +43,16 @@ def accessible_attributes
 
         # Used by the migration template to determine the parent name of the model
         def parent_class_name
-          options[:parent] || "ActiveRecord::Base"
+          options[:parent] || determine_default_parent_class
         end
 
+        def determine_default_parent_class
+          if File.exist?('app/models/application_record.rb')
+            "ApplicationRecord"
+          else
+            "ActiveRecord::Base"
+          end
+        end
```

Comment:
Wonder if we could always generate with `ApplicationRecord` superclass and provide a fallback (perhaps via const_missing? alias to AR::Base) and warning message when the app doesn't have an `app/models/application_record.rb`